framework/platform/bpl/linux/bpl_cfg.cpp

```diff
@@ -83,20 +83,18 @@ int cfg_is_enabled() { return 1; }
 
 int cfg_is_master()
 {
-    std::string mode_str;
-    if (cfg_get_param("management_mode=", mode_str) < 0) {
-        MAPF_ERR("cfg_is_master: Failed to read ManagementMode");
-        return RETURN_ERR;
-    }
-
-    if (mode_str == "Multi-AP-Controller-and-Agent" || mode_str == "Multi-AP-Controller")
+    switch (cfg_get_management_mode()) {
+    case BPL_MGMT_MODE_MULTIAP_CONTROLLER_AGENT:
         return 1;
-    else if (mode_str == "Multi-AP-Agent")
+    case BPL_MGMT_MODE_MULTIAP_CONTROLLER:
+        return 1;
+    case BPL_MGMT_MODE_MULTIAP_AGENT:
         return 0;
-
-    MAPF_ERR("cfg_is_master: Unexpected management_mode " << mode_str);
-
-    return RETURN_ERR;
+    case BPL_MGMT_MODE_NOT_MULTIAP:
+        return (cfg_get_operating_mode() == BPL_OPER_MODE_GATEWAY) ? 1 : 0;
```

Comment:
Hm, it seems that in MULTIAP mode, the operating mode is redundant then... Actually, AFAICS, the operating mode is only used to determine if we're a gateway or not. So maybe that should be encoded here as well?

Anyway can be handled later.

Reply:
Agree - there is a lot of redundancy here, I actually don't think we need all the modes, and indeed - it can be handled later.